doc/source/walkthrough.rst

```diff
@@ -215,131 +404,282 @@ Object IDs can be created in multiple ways.
-    def long_running_function()
-        time.sleep(8)
+    .. autofunction:: ray.get
+        :noindex:
 
-    obj_id = long_running_function.remote()
-    try:
-        ray.get(obj_id, timeout=4)
-    except RayTimeoutError:
-        print("`get` timed out.")
+  .. group-tab:: Java
 
-.. autofunction:: ray.get
-    :noindex:
+    The ``RayObject::get`` method is used to fetch a single object
+    from the corresponding remote object. If the current node's object store
+    does not contain the object, the object is downloaded.
 
+    If you want to fetch multiple remote objects, you can use the
+    ``Ray.get`` method, which will fetch multiple objects in parallel.
+
+    .. code:: java
+
+      List<RayObject<Integer>> rayObjectList = new ArrayList<>();
+      for (int i = 0; i < 4; i++) {
+        rayObjectList.add(Ray.call(MyRayApp::foo));
+      }
+      // Fetch multiple objects in parallel.
+      List<Integer> results = Ray.get(rayObjectList);
+      Assert.assertEquals(results, ImmutableList.of(1, 1, 1, 1));
+
+    If an exception occurs while executing a remote function, the exception
+    will be stored in the ``RayObject``. When fetching the result, a `RayTaskException`
+    will be thrown.
+
+    .. code:: java
+
+      public static int badMethod() {
+        throw new RuntimeException("An expected exception occurred.");
+      }
+
+      try {
+        rayObject.get();
+      } catch (RayTaskException e) {
+        Throwable rootCause = e.getCause();
+        Assert.assertTrue(rootCause instanceof RuntimeException);
+      }
 
 After launching a number of tasks, you may want to know which ones have
-finished executing. This can be done with ``ray.wait``. The function
+finished executing. This can be done with ``wait``. The function
 works as follows.
 
-.. code:: python
+.. tabs::
+  .. group-tab:: Python
+
+    .. code:: python
+
+        ready_ids, remaining_ids = ray.wait(object_ids, num_returns=1, timeout=None)
+
+    .. autofunction:: ray.wait
+        :noindex:
+
+  .. group-tab:: Java
 
-    ready_ids, remaining_ids = ray.wait(object_ids, num_returns=1, timeout=None)
+    .. code:: java
 
-.. autofunction:: ray.wait
-    :noindex:
+        WaitResult<Integer> waitResult = Ray.wait(rayObjectList, /*num_returns=*/1, /*timeoutMs=*/1000);
+        System.out.println(waitResult.getReady());  // List of ready objects.
+        System.out.println(waitResult.getUnready());  // list of unready objects.
 
 Object Eviction
 ---------------
 
-When the object store gets full, objects will be evicted to make room for new objects.
-This happens in approximate LRU (least recently used) order. To avoid objects from
-being evicted, you can call ``ray.get`` and store their values instead. Numpy array
-objects cannot be evicted while they are mapped in any Python process. You can also
-configure `memory limits <memory-management.html>`__ to control object store usage by
-actors.
+.. tabs::
 
-.. note::
+  .. group-tab:: Python
```

Comment:
Since the semantics are the same here, it's just minor changes to the API calls, can we just add a code block for the call as opposed to changing the whole paragraph?

Having to maintain huge groups of text between the two is going to be very difficult.